sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -527,6 +540,12 @@ public void processElement(ProcessContext ctx) {
                 }
               }));
     }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+      read.populateDisplayData(builder);
```

Comment:
I don't think it matters. 
IMHO with/without metadata is more for the pipeline author and what he does with the read PCollection. DisplayData for the KafkaIO should be about `T` which can be `KV<K, V>` or `KafkaRecord<KV<K, V>>`